stdlib/public/core/String.swift

```diff
@@ -968,3 +968,73 @@ internal func _fastWithNormalizedCodeUnitsImpl(
     }
   }
 }
+
+// Allow for Swift String class to convert itself to int/float/double using dot notation
+// This will allow for an easier time when casting a string to various types of numbers
+// Since it returns a non-optional value, unwrapping isn't necessary
+// Before we would have to do Float(value) ?? 0.0, where as now it would be simply value.intValue etc.
+extension String {
+    func intValue() -> Int {
+        if let value = Int(self) {
+            return value;
+        }
+        else {
+            return 0;
```

Comment:
These functions would have to return `nil` on failure. What if the user passes like "FOO BAR"? That's not a valid integer value, so returning `0` is incorrect. 

Also, if you're returning `nil`, then the whole idea of this function is moot because there's no difference between `"123".intValue() ?? <some default value>` vs `Int("123") ?? <some default value>`.

Reply:
Well my thoughts were why did the Obj-c language have these functions for NSString? Where as now Swift uses the String class which doesn't allow for these conversions.

I personally liked that it returned 0, 0.0 when the string wasn't a valid number, i don't want to have to deal with optionals in this case. Returning 0 or 0.0 or better then returning nil and having to handle this case. But that's just my opinion. 

i currently use this particular extension in all of my current Swift projects to save me the hassle of having to handle the null case when using Float(string) or Double(string) which returns optional values.

Reply:
How can the user of this new function distinguish between `0` being returned when the string is invalid (eg: “not zero”) and `0` returned when the string is “0”?

Reply:
I see what you're saying. My intention for this extension was to mimic the NSString functions for integerValue, floatValue, and doubleValue.

Reply:
Hmm I suppose you can add some documentation to say that the function will return `0` if the string is invalid, so users would know that the above behaviour is expected.

Reply:
Ye so it looks like i need to actually do a couple things before going any further here. 1. Add public to the function titles. 2. Better documentation

We can continue this discussion afterwards. 

I appreciate the conversation and hope to continue it at a later time.

Reply:
I recommend starting a conversation on the Swift forums (under Evolution) so the community can discuss this in more depth: https://forums.swift.org/c/evolution